hudi-common/src/main/java/org/apache/hudi/common/bootstrap/index/HFileBootstrapIndex.java

```diff
@@ -200,582 +166,4 @@ public void dropIndex() {
-          HoodieBootstrapPartitionMetadata metadata =
-              TimelineMetadataUtils.deserializeAvroMetadata(valBytes, HoodieBootstrapPartitionMetadata.class);
-          return metadata.getFileIdToBootstrapFile().entrySet().stream()
-              .map(e -> new BootstrapFileMapping(bootstrapBasePath, metadata.getBootstrapPartitionPath(),
-                  partition, e.getValue(), e.getKey())).collect(Collectors.toList());
-        } else {
-          LOG.warn("No value found for partition key (" + partition + ")");
-          return new ArrayList<>();
-        }
-      } catch (IOException ioe) {
-        throw new HoodieIOException(ioe.getMessage(), ioe);
-      }
-    }
-
-    @Override
-    public String getBootstrapBasePath() {
-      return bootstrapBasePath;
-    }
-
-    @Override
-    public Map<HoodieFileGroupId, BootstrapFileMapping> getSourceFileMappingForFileIds(
-        List<HoodieFileGroupId> ids) {
-      Map<HoodieFileGroupId, BootstrapFileMapping> result = new HashMap<>();
-      // Arrange input Keys in sorted order for 1 pass scan
-      List<HoodieFileGroupId> fileGroupIds = new ArrayList<>(ids);
-      Collections.sort(fileGroupIds);
-      try {
-        HFileReader reader = fileIdIndexReader();
-        reader.seekTo();
-        for (HoodieFileGroupId fileGroupId : fileGroupIds) {
-          Key lookupKey = new UTF8StringKey(getFileGroupKey(fileGroupId));
-          if (reader.seekTo(lookupKey) == HFileReader.SEEK_TO_FOUND) {
-            org.apache.hudi.io.hfile.KeyValue keyValue = reader.getKeyValue().get();
-            byte[] valBytes = IOUtils.copy(
-                keyValue.getBytes(), keyValue.getValueOffset(), keyValue.getValueLength());
-            HoodieBootstrapFilePartitionInfo fileInfo = TimelineMetadataUtils.deserializeAvroMetadata(valBytes,
-                HoodieBootstrapFilePartitionInfo.class);
-            BootstrapFileMapping mapping = new BootstrapFileMapping(bootstrapBasePath,
-                fileInfo.getBootstrapPartitionPath(), fileInfo.getPartitionPath(), fileInfo.getBootstrapFileStatus(),
-                fileGroupId.getFileId());
-            result.put(fileGroupId, mapping);
-          }
-        }
-      } catch (IOException ioe) {
-        throw new HoodieIOException(ioe.getMessage(), ioe);
-      }
-      return result;
-    }
-
-    @Override
-    public void close() {
-      try {
-        if (indexByPartitionReader != null) {
-          indexByPartitionReader.close();
-          indexByPartitionReader = null;
-        }
-        if (indexByFileIdReader != null) {
-          indexByFileIdReader.close();
-          indexByFileIdReader = null;
-        }
-      } catch (IOException ioe) {
-        throw new HoodieIOException(ioe.getMessage(), ioe);
-      }
-    }
-  }
-
-  /**
-   * HBase HFile reader based Index Reader.  This is deprecated.
-   */
-  public static class HBaseHFileBootstrapIndexReader extends BootstrapIndex.IndexReader {
```

Comment:
Can we keep this reader, now that we can use reflection and config to choose which bootstrap index reader to use?

Reply:
put it into it's own file